lib/src/signal_client.dart

```diff
@@ -180,42 +240,53 @@ class SignalClient {
     _ws?.sink.add(buf);
   }
 
-  void _handleMessage(dynamic message) {
-    if (message is! List<int>) {
-      return;
-    }
-    final msg = SignalResponse.fromBuffer(message);
-    switch (msg.whichMessage()) {
-      case SignalResponse_Message.join:
-        if (!_connected) {
-          _connected = true;
-          delegate?.onConnected(msg.join);
-        }
-        break;
-      case SignalResponse_Message.answer:
-        delegate?.onAnswer(toRTCSessionDescription(msg.answer));
-        break;
-      case SignalResponse_Message.offer:
-        delegate?.onOffer(toRTCSessionDescription(msg.offer));
-        break;
-      case SignalResponse_Message.trickle:
-        delegate?.onTrickle(toRTCIceCandidate(msg.trickle.candidateInit), msg.trickle.target);
-        break;
-      case SignalResponse_Message.update:
-        delegate?.onParticipantUpdate(msg.update.participants);
-        break;
-      case SignalResponse_Message.trackPublished:
-        delegate?.onLocalTrackPublished(msg.trackPublished);
-        break;
-      case SignalResponse_Message.speaker:
-        delegate?.onActiveSpeakersChanged(msg.speaker.speakers);
-        break;
-      case SignalResponse_Message.leave:
-        delegate?.onLeave(msg.leave);
-        break;
-      default:
-        log('unsupported message: ' + json.encode(msg));
-    }
+  Future<void> _handleMessage(dynamic message) async {
+    //
+    if (message is! List<int>) return;
+
+    final msg = lk_rtc.SignalResponse.fromBuffer(message);
+
+    await lock.synchronized(() async {
```

Comment:
I *think* _handleMessage is already called in a synchronous fashion. though it doesn't hurt to ensure that.

Reply:
I think the delegates are getting fired in the middle of another delegate. So this makes it wait for the delegate to finish before firing the next handler.